docs/source/format/FormatIntro.rst

```diff
@@ -0,0 +1,511 @@
+and the nullness is determined by the child arrays.
+
+Arrow defines two distinct union types, “dense” and “sparse”.
+
+Dense Union
+^^^^^^^^^^^
+
+Dense Union has one child array for each type present in the mixed-type array and
+
+* **Types buffer:** holds type id for each slot of the array. Type id corresponds
+  to the number of the child array.
+* **Offsets buffer:** holds relative offset into the respective child array for each
+  array slot.
+
+.. figure:: ./images/dense-union-diagram.svg
+   :alt: Diagram is showing the difference between the dense union data type
+         presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for dense union data type.
+
+Sparse union
+^^^^^^^^^^^^
+
+A sparse union has the same structure as a dense union, with the omission of the offsets
+array. In this case, the child arrays are each equal in length to the length of the union.
+
+
+.. figure:: ./images/sparse-union-diagram.svg
+   :alt: Diagram is showing the difference between the sparse union data type
+         presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for sparse union data type.
+
+Dictionary Encoded Layout
+=========================
+
+Dictionary encoding can be effective when you have data with many repeated values.
+The values are represented by integers referencing a dictionary usually consisting of
+unique values.
+
+.. figure:: ./images/dictionary-diagram.svg
+   :alt: Diagram is showing the difference between the dictionary data type
+         presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for dictionary data type.
+
+Run-End Encoded Layout
+======================
+
+Run-end encoding is well-suited for representing data containing sequences of the
+same value. These sequences are called runs. Run-end encoded array has no buffers
+by itself, but has two child arrays:
+
+*  **Run ends array:** holds the index in the array where each run ends.
+*  **Values array:** the actual values without repetitions.
+
+.. figure:: ./images/ree-diagram.svg
+   :alt: Diagram is showing the difference between the run-end encoded data
+         type presented in a Table and the dataactually stored in computer
+         memory.
+
+   Physical layout diagram for run-end encoded data type.
+
+.. link to All types overview https://github.com/apache/arrow/issues/14752
+
+Extension Types
+===============
+
+In case the system or application needs to extend standard Arrow data types with
+custom semantics this is enabled by defining extension types or user-defined types.
+
+For example:
+
+* Universally unique identifier (uuid) can be represented as a FixedSizeBinary type
+* Trading time can be represented as a Timestamp with metadata indicating the market
+  trading calendar
+
+Extension types can be defined by annotating any of the built-in Arrow logical types
```

Comment:
In https://github.com/apache/arrow/issues/14752 we were trying to get away from using the term "logical type" where it's not technically correct. Columnar.rst still needs to be updated but I think avoiding propagating the term here would be good.
```suggestion
Extension types can be defined by annotating any of the built-in Arrow data types
```